src/Covenant/Analysis/Poetry/PoetryHashParser.cs

```diff
@@ -0,0 +1,30 @@
+namespace Covenant.Analysis.Npm;
+
+internal static class PoetryHashParser
+{
+    public static BomHash? Parse(string? text)
+    {
+        if (string.IsNullOrWhiteSpace(text))
+        {
+            return null;
+        }
+
+        var index = text.IndexOf(':');
+        if (index == -1)
+        {
+            // Unknown hash format
```

Comment:
Is this considered normal? This sounds like it might warrant throwing an exception. (And then you'd make the return type `string`, not `string?`)